benchmarks/nemo/nemolite2d/manual_versions/psykal_c_opencl/time_step_mod.f90

```diff
@@ -140,30 +144,32 @@ SUBROUTINE invoke_time_step(istp, ssha_t, ssha_u, ssha_v, sshn_t, sshn_u, &
             g &
         )
 
-        ! Mark data_on_device flags
-        ssha_t%data_on_device = .true.
-        sshn_t%data_on_device = .true.
-        sshn_u%data_on_device = .true.
-        sshn_v%data_on_device = .true.
-        hu%data_on_device = .true.
-        hv%data_on_device = .true.
-        un%data_on_device = .true.
-        vn%data_on_device = .true.
-        ua%data_on_device = .true.
-        ht%data_on_device = .true.
-        ssha_u%data_on_device = .true.
-        va%data_on_device = .true.
-        ssha_v%data_on_device = .true.
+        if (first_time) then
+            ! Mark data_on_device flags
```

Comment:
Do you need a `first_time = .false.` here?

Reply:
It was at the end of the condition, but I moved it here now for clarity.

Reply:
Ops no, it wasn't there :)